crates/nu_plugin_parse/src/nu/mod.rs

```diff
@@ -142,77 +111,49 @@ fn parse(input: &str) -> Vec<ParseCommand> {
         }
 
         if !before.is_empty() {
-            output.push(ParseCommand::Text(before.to_string()));
+            output.push_str(&regex::escape(&before));
         }
+
         // Look for column as we're now at one
         let mut column = String::new();
-
         while let Some(c) = loop_input.next() {
             if c == '}' {
                 break;
             }
             column.push(c);
+
+            if loop_input.peek().is_none() {
+                return Err(ShellError::labeled_error(
+                    "Found opening `{` without an associated closing `}`",
+                    "invalid parse pattern",
+                    tag,
+                ));
+            }
         }
 
         if !column.is_empty() {
-            output.push(ParseCommand::Column(column.to_string()));
+            output.push_str("(?P<");
+            output.push_str(&column);
+            output.push_str(">.+?)");
         }
 
         if before.is_empty() && column.is_empty() {
             break;
         }
     }
 
-    output
-}
-
-impl From<&[ParseCommand]> for ColumnNames {
-    fn from(commands: &[ParseCommand]) -> ColumnNames {
-        let mut output = vec![];
-
-        for command in commands {
-            if let ParseCommand::Column(c) = command {
-                output.push(c.clone());
-            }
-        }
-
-        ColumnNames(output)
-    }
-}
-
-impl From<&Regex> for ColumnNames {
-    fn from(regex: &Regex) -> ColumnNames {
```

Comment:
We now build a regex with the capture group names when using the "simple form" of a parse expression, so we only need this codepath. Factored out into a function.